File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:

    bool hasDuplicate(vector<int>& nums) {
        int flag = 0;
        if(nums.size()==0)
        {
            return false;
        }
        for(int i = 0;i<nums.size()-1;i++)
        {
            for(int j=i+1;j<nums.size();j++)
            {
                // cout<<"array i: "<<nums[i]<<endl;
                // cout<<"array j: "<<nums[j]<<endl;
                if(nums[i]==nums[j])
                {
                    flag = 1;
                    return true;
                }
            }
        }

        if (flag==0)
        {
            return false;
        }
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        total=1
        ans=[]
        flag=0
        if nums.count(0) > 1:
            for i in range(len(nums)):
                ans.append(0)
            return ans
        
        for tmp in nums:
            if tmp == 0:
                flag=1
                continue
            else:
                total *= tmp
        for i in range(len(nums)):
            if nums[i]==0:
                ans.append(total)
            elif flag==1:
                ans.append(0)
            else:
                ans.append(int(total / nums[i]))
        return ans